assets/scripts/palette/__tests__/UndoRedo.test.js

```diff
@@ -1,98 +1,103 @@
 /* eslint-env jest */
 import React from 'react'
-import { mountWithIntl as mount } from '../../../../test/helpers/intl-enzyme-test-helper.js'
-import { UndoRedo } from '../UndoRedo'
-import { isUndoAvailable, isRedoAvailable } from '../../streets/undo_stack'
-
-jest.mock('../../streets/undo_stack')
-jest.mock('../../store/actions/undo', () => ({
-  undo: jest.fn(),
-  redo: jest.fn()
+import { cleanup, fireEvent } from '@testing-library/react'
+import { renderWithReduxAndIntl } from '../../../../test/helpers/render'
+import UndoRedo from '../UndoRedo'
+import { replaceUndoStack } from '../../store/actions/undo'
+
+// `getRemixOnFirstEdit` is a legacy function, so for the purposes of this
+// test suite it always returns `false` (aka, assume user owns the street)
+jest.mock('../../streets/remix', () => ({
+  getRemixOnFirstEdit: () => false
 }))
 
 describe('UndoRedo', () => {
-  it('renders two buttons', () => {
-    const wrapper = mount(<UndoRedo locale={{}} />)
-    expect(wrapper.find('button').length).toEqual(2)
+  afterEach(() => {
+    cleanup()
   })
 
-  it('checks if undo or redo is available when undo position changes', () => {
-    // `<UndoRedo>` is mounted with initial props
-    const wrapper = mount(<UndoRedo undoPosition={0} undoStack={[]} />)
+  it('renders two buttons', () => {
+    const wrapper = renderWithReduxAndIntl(<UndoRedo />)
+    expect(wrapper.asFragment()).toMatchSnapshot()
+  })
 
-    // Send new props to `<UndoRedo>`
-    // It doesn't really matter what we set props to.
-    wrapper.setProps({
-      undoPosition: 1,
-      undoStack: []
+  it('enables undo button when an undo stack is created', () => {
+    const wrapper = renderWithReduxAndIntl(<UndoRedo />, {
+      initialState: {
+        undo: {
+          stack: [],
+          position: 0
+        }
+      }
     })
+    const undoButton = wrapper.getByTitle('Undo')
+    const redoButton = wrapper.getByTitle('Redo')
 
-    expect(isUndoAvailable).toBeCalled()
-    expect(isRedoAvailable).toBeCalled()
+    // Initial button states are disabled
+    expect(undoButton.disabled).toEqual(true)
+    expect(redoButton.disabled).toEqual(true)
 
-    // Clear mocks for future tests
-    isUndoAvailable.mockClear()
-    isRedoAvailable.mockClear()
-  })
+    // Update undo stack state from "elsewhere", and `<UndoRedo>` should
+    // receive new props. This is akin to a user editing the street once
+    wrapper.store.dispatch(replaceUndoStack([{ foo: 'bar' }], 1))
```

Comment:
I do want to test this change after the initial mount. The idea is that the undo/redo buttons update their appearance if the store changes. Is this something that's worth testing at all?

Reply:
I'll just completely remove this test because what I'm really testing is that the component updates... and it does, in other tests. The source of the update probably isn't important and we can remove having a dispatch in the test.